packages/block-editor/src/store/actions.js

```diff
@@ -1432,36 +1439,128 @@ export const setNavigationMode =
  */
 export const __unstableSetEditorMode =
 	( mode ) =>
-	( { dispatch, select } ) => {
-		// When switching to zoom-out mode, we need to select the root block
-		if ( mode === 'zoom-out' ) {
-			const firstSelectedClientId = select.getBlockSelectionStart();
-			if ( firstSelectedClientId ) {
-				dispatch.selectBlock(
-					select.getBlockHierarchyRootClientId(
+	( { dispatch, select, registry } ) => {
+		registry.batch( () => {
+			// When switching to zoom-out mode, we need to select the root block
+			if ( mode === 'zoom-out' ) {
+				const firstSelectedClientId = select.getBlockSelectionStart();
+				if ( firstSelectedClientId ) {
+					const rootClientId = select.getBlockHierarchyRootClientId(
 						firstSelectedClientId
-					)
-				);
+					);
+					dispatch.selectBlock( rootClientId );
+				}
+			}
+
+			const prevMode = select.__unstableGetEditorMode();
+
+			if ( mode === 'zoom-out' || prevMode === 'zoom-out' ) {
```

Comment:
I refactored things to group the toggle on/off together for each branch of `sectionsContainerClientId` instead of having two `sectionsContainerClientId` checks. I think the code reads better that way, but it means I needed to add an extra layer so we don't run a bunch of random unused filters and maps in the `else` branch inside here.